plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/HostedMachineProviderImpl.java

```diff
@@ -215,4 +267,22 @@ protected void buildImage(CheServiceImpl service,
             }
         }
     }
+
+    @PreDestroy
+    private void preDestroy() {
+        finalizeSnapshotImagesCleaner();
+    }
+
+    /**
+     * Tries to clean up snapshots images which is in the cleaner queue, but were canceled due to server stopping.
+     * To avoid freezing of server on stop performs post clean in separate thread.
+     */
+    private void finalizeSnapshotImagesCleaner() {
```

Comment:
I do not think so. Separate method group logic, provides method docs, and from its name someone can quickly figure out what that code does.